controllers/api/v1/health_check_test.go

```diff
@@ -1,41 +1,94 @@
 package apiv1
 
 import (
+	"fmt"
 	"io/ioutil"
 	"net/http"
 
+	_ "google-scraper/initializers"
 	. "google-scraper/tests"
 
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("HealthCheckController", func() {
 	Describe("GET /api/v1/health-check", func() {
-		It("returns 200 status code", func() {
-			response := MakeRequest("GET", "/api/v1/health-check", nil)
+		Context("Given valid credentials", func() {
+			It("returns 200 status code", func() {
+				email := "john@example.com"
+				password := "secret"
+				user := FabricateUser("John", email, password)
+				client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+				oauthToken := FabricateOAuthToken(client, user.Id)
+				header := http.Header{"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())}}
 
-			Expect(response.StatusCode).To(Equal(http.StatusOK))
-		})
+				response := MakeAuthenticatedRequest("GET", "/api/v1/health-check", header, nil, nil)
+
+				Expect(response.StatusCode).To(Equal(http.StatusOK))
+			})
 
-		It("returns correct JSON response", func() {
-			expectedResponse := `{
-				"data": {
-					"type": "health_check",
-					"id": "0",
-					"attributes": {
-						"success": true
+			It("returns correct JSON response", func() {
+				email := "john@example.com"
+				password := "secret"
+				user := FabricateUser("John", email, password)
+				client := FabricateOAuthClient(uuid.New().String(), uuid.New().String())
+				oauthToken := FabricateOAuthToken(client, user.Id)
+				header := http.Header{"Authorization": {fmt.Sprintf("Bearer %v", oauthToken.GetAccess())}}
+				expectedResponse := `{
+					"data": {
+						"type": "health_check",
+						"id": "0",
+						"attributes": {
+							"success": true
+						}
 					}
+				}`
+
+				response := MakeAuthenticatedRequest("GET", "/api/v1/health-check", header, nil, nil)
+
+				responseBody, err := ioutil.ReadAll(response.Body)
+				if err != nil {
+					Fail("Reading response body failed: " + err.Error())
 				}
-			}`
 
-			response := MakeRequest("GET", "/api/v1/health-check", nil)
-			responseBody, err := ioutil.ReadAll(response.Body)
-			if err != nil {
-				Fail("Reading response body failed: " + err.Error())
-			}
+				Expect(string(responseBody)).To(MatchJSON(expectedResponse))
+			})
+		})
+
+		Context("Given INVALID credentials", func() {
+			It("returns 401 status code", func() {
+				header := http.Header{"Authorization": {fmt.Sprintf("Bearer %v", "invalid")}}
+
+				response := MakeAuthenticatedRequest("GET", "/api/v1/health-check", header, nil, nil)
```

Comment:
@junan Won't you make this endpoint as public? 🤔
I think it'd be nice if it's public for anyone can check the website's status

Reply:
In this article it suggested to make the health check endpoint  private and only accessible internally 
https://blog.runscope.com/posts/build-a-health-check-api-for-services-underneath-your-api

![Screen Shot 2021-04-07 at 1 39 38 PM](https://user-images.githubusercontent.com/6076762/113821812-cd51fd80-97a6-11eb-9b62-fdb509c95066.png)

Reply:
Updated here https://github.com/junan/google-scraper/pull/51/commits/b1aee2918a904c2bf8b0e732d1e30a5d6a072063